app/controllers/api/v1/task_controller.rb

```diff
@@ -0,0 +1,89 @@
+class Api::V1::TaskController < ApplicationController
+  before_action :require_admin, only: [:delete, :update, :show_all_by_sort, :find_by_category]
+
+  def create
+    service = TaskService.new
+    result = service.create(current_user, task_params)
+    if service.errors.present?
+      render json: { error: service.errors }, status: :unprocessable_entity
+    else
+      render json: { message: 'Task Assigned Successfully' }, status: 200
+    end
+  end
+
+  def delete
+    service = TaskService.new
+    result = service.delete(params[:id])
+    if service.errors.present?
+      render json: { error: service.errors }, status: :unprocessable_entity
+    else
+      render json: { message: 'Task Deleted Successfully' }, status: 200
+    end
+  end
+
+  def index 
+    service = TaskService.new
+    tasks = service.find_all(current_user)
+    if service.errors.present?
+      render json: { error: service.errors }, status: :unprocessable_entity
+    elsif tasks.blank?
+      render json: { message: 'Tasks List is Empty' }, status: :ok
+    else
+      render json: tasks, each_serializer: TaskSerializer, status: :ok
+    end
+  end
+
+  def show_all_by_sort
+    service = TaskService.new
+    tasks = service.show_all_by_sort(params[:field])
+    if service.errors.present?
+      render json: { error: service.errors }, status: :unprocessable_entity
+    elsif tasks.blank?
+      render json: { message: 'Tasks List is Empty' }, status: :ok  
+    else
+      render json: tasks, each_serializer: TaskSerializer, status: :ok
+    end
+  end
```

Comment:
this should not be present as a separate API, move sorting logic to index API, use params to find what sorting is needed.